services_app/src/androidTest/java/org/opendatakit/database/service/GroupsPermissionTest.java

```diff
@@ -193,8 +190,8 @@ private ContentValues contentValuesTestSeti(int i, String groupReadOnly, String
    private void verifyRowTestSeti(TypedRow row, int i, String groupReadOnly, String groupModify,
        String groupPrivileged) {
       assertEquals(row.getDataByKey(COL_STRING_ID), TEST_STR_i + i);
-      assertEquals(row.getDataByKey(COL_INTEGER_ID), Long.valueOf(TEST_INT_i + i));
```

Comment:
Not sure the conversion needs removing. There a bunch of casts going on otherwise and sometimes explicit code is better. 
I just checked and and Long.valueOf is NOT deprecated so it is correct to use.
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Long.html

Additiaonlly adding an integer and integer together produce an integer. The test is trying to verify a double is stored correctly. Therefore you are changing the test!!!!